### PR TITLE
Remove setup log.

### PR DESCRIPTION
**Pull request by ConnorRigby**
This function logs wifi settings which can
contain wifi credentials.

### CODE REVIEW

File: lib/nerves_network.ex [hunk 1]
```diff
@@ -48,7 +48,7 @@ defmodule Nerves.Network do
   """
   @spec setup(Types.ifname(), setup_settings) :: :ok
   def setup(ifname, settings \\ []) do
-    Logger.debug("#{__MODULE__} setup(#{ifname}, #{inspect(settings)})")
+    Logger.debug("#{__MODULE__} setup(#{ifname})")
```
Comment (fhunleth) [inline lib/nerves_network.ex]: Does this debug log provide any value any more? It seems like without the settings that it's not very useful.

Comment (fhunleth) [reply lib/nerves_network.ex]: I'm not sure. Given how hard networking is, my leaning would be to leave the log message, but override the WPA key with "*****" if it exists.

Comment (ConnorRigby) [reply lib/nerves_network.ex]: I thought about doing this, but i had some concerns.

1) `settings` can be a map, or a keyword list. (this is an ongoing issue)
2) wpa_supplicant supports quite a few different credentials that should be considered sensitive.
   * `PSK`
   * `wep_key0..3`
   * Some other things for WPA_EAP that i don't know off the top of my head. 

a Possible solution for 1 could be that we move this log to right before `<manager>.start_link/2` is called inside `Nerves.Network.Config`, where the settings have been converted to a map.

Comment (fhunleth) [reply lib/nerves_network.ex]: Ok, then let's punt this decision to our future selves when we have an actual use for it.

**Pull request merged**